src/vs/workbench/parts/files/browser/editors/textFileEditor.ts

```diff
@@ -67,6 +70,12 @@ export class TextFileEditor extends BaseTextEditor {
 		}
 	}
 
+	private onWillCloseEditor(e: IEditorIdentifier): void {
+		if (e.editor === this.input && this.position === this.editorGroupService.getStacksModel().positionOfGroup(e.group)) {
```

Comment:
It is fine to do the equality check here between editors here? They are actually the same reference?
I understand you do not have an `id` for the editors here.